app/src/main/java/com/architectcoders/aacboard/ui/fragments/EditBoardCellFragment.kt

```diff
@@ -1,6 +1,47 @@
 package com.architectcoders.aacboard.ui.fragments
 
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
 import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
 import com.architectcoders.aacboard.R
+import com.architectcoders.aacboard.data.PictogramUI
+import com.architectcoders.aacboard.databinding.FragmentEditBoardCellBinding
+import com.architectcoders.aacboard.ui.utils.getNavigationResultLiveData
+import com.architectcoders.aacboard.ui.utils.loadUrl
 
-class EditBoardCellFragment : Fragment(R.layout.fragment_edit_board_cell)
\ No newline at end of file
+class EditBoardCellFragment : Fragment(R.layout.fragment_edit_board_cell){
+    private var _binding: FragmentEditBoardCellBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        _binding = FragmentEditBoardCellBinding.inflate(inflater, container, false)
+        initViews()
+        val result= getNavigationResultLiveData<PictogramUI>()
+        result?.observe(viewLifecycleOwner) { pictogram -> updateUI(pictogram) }
+        return binding.root
+    }
+
+    private fun initViews() {
+        binding.apply {
+            pictogram.setOnClickListener {
+                findNavController().navigate(R.id.action_editBoardCell_to_searchPictograms)
+            }
+
+        }
+    }
+
+    private fun updateUI(pictogram: PictogramUI?) {
+        pictogram?.let {
+            binding.pictogram.loadUrl(it.url)
+            binding.keyword.setText(it.keyword)
+        }
+    }
+
+}
```

Comment:
Not sure about not having a viewModel here. I think viewModels can access to the saveStateHandle, if so my approach would be to have a viewModel that fetches the data from the state handle and updates a flow which is collected by the fragment. Here the view is responsible for many things, fetching the data and updating the UI.

We can review/investigate this together if you want.

Reply:
There is a ViewModel in the final version. In this PR EditBoardCellFragment was just created to trigger SearchPictogramsFragment and validate that it received the right response when a pictogram was selected.

If we move this logic to ViewModel, we need to:
- add saveStateHandle in ViewModel constructor (and Koin module)
- observe the livedata (or rather StateFlow as you propose) in viewModel 

I will probably need help to get it done but I am going to try.